### ISSUE TITLE
720p doesn't buffer anymore after installing certain other youtube addons

### ISSUE THREAD

**Issue by dreamcat4** [Kolifanes/plugin.video.youtube#53]
Hi,
Youtube plugin used to be working fine for me on 720p. I can also watch 720p videos in Browser just fine.

Kodi 16.1 Jarvis (Git:20160424-c327c53)
Youtube (v5.2.1)

Then I installed these 2 addons:

youtube-dl Control (v16.627.0)
YouTube Download Context Menu (v1.1.0)

And after installing the 2nd addon, 720p no longer plays well. There is some buffering issue.

**_360p and lower quality:**_
- videos still play just fine

**_720p:**_
- Video starts paused
- Says "Buffering 1%..." and buffering just continues forever, until the cache size is 100% used up
- Pressing play will start playing the video immediately

BUT
- Then once buffer end is reached, sound cuts out and video scrambles
- Its supposed to auto-pause at this point, and buffer a little more. Doesnt do that.
- Have to manually pause / un-pause and intervene when buffer is reached.

THEN:
- disabled the extra youtube addons, did not solve anything
- un-installed the extra youtube addons, still issue remains

So what could be the issue here? Is there some way I can clear away these addons or associated data and start again? Which addon is responsible? Perhaps it is some addon dependancy shared library, or of different version, which got installed, but was not removed again?

**Comment by dreamcat4** [comment]
Some more points:
- This issue also described / reported same by other kodi forum users.
- Seems like this addon relies upon Youtube HTML5 javascript embedded player. So maybe there is another place on the system. Somewhere else to clear the browser javascript cache of it? I cannot find
- While 'Buffering...', the download isn't using the full connection speed. My WAN link is 3.2 mbit, but this downloading is strangely self-limited itself to 1.2 mbit / sec?

![](http://i.imgur.com/E7filV6.png)

**Comment by jdf76** [comment]
What are you specs, Windows 8, 8.1, or 10? are you using a laptop? I don't have issues with my mac mini and kodi. You may want to hit up the forum http://forum.kodi.tv/showthread.php?tid=267160

**Comment by dreamcat4** [comment]
Ah hi there! Yeah... was just going to make a forum post asking about it soon enough.

Yeah sorry So missing  OS / HW specs. Am using:
- Win10 Pro 64-bit
- i5-6600K,  4.6Ghz
- SanDisk Extreme II SSD 480GB

Another thing:

Did look inside my `kodi.log` file while its 'Buffereing 1%...'. But nothing new or relevant appears in the default log file.

Like I said before... this youtube plugin has been working fine for me several weeks @ 720p. It was only after installing some extra 'youtube-dl Contol' plugin did it break in this way. So it seems a lot like some software related thing to me.

In time might eventually get around to fresh / clean install everything from scratch. All of it. But probably not for a while yet.

**Comment by dreamcat4** [comment]
Youtube videos also still continue play OK fine at 720p setting in Firefox, Chrome and Edge Browsers.

**Comment by dreamcat4** [comment]
Hi again. Isn't this symptom that the youtube DASH has become broken on my system?

**Comment by h0d3nt3uf3l** [comment]
Do you use the mpeg dash support from andxpanic?
Its not tested and implemented very well.

**Comment by dreamcat4** [comment]
Hi! Yeah sorry I am not familiar enough with this plugins landscape. But that would explain things pretty well... its just those plugins listed (and versions) from their respective sources.

I look in the plugin `youtube-dl control` and it uses a python lib called `youtube-dl` underneath and so many python files.

Want to remove those other 2 plugins the main youtube playback will work better. It just doesnt seem to clear state after un-installing them. The poor playback continue to happen. Maybe if I knew which files delete those other dependant python libs and / or whatever hidden cache setting which still exists and remains after un-installing those other topmost user plugin.

**Comment by dreamcat4** [comment]
Hmm OK. On a fresh windows install, all new installation. It did not yet reproduce the same issue. And with installed these extra youtube 'helper' addons. Even after enabling the 2 most questionable looking settings, such as 'support alternative player' (youtube) and 'disable dash' (youtube-dl control). The 720p playback is still fine on the new system.

Meanwhile back on original system, download speed still artificially limited and remains consistently too slow for doing 720p playback.

Really do not know what to check / where to look for the difference. I'm suspiciously convinced there must be another browser cache somewhere else. For the embedded HTML5 javascript player. Which has gone bad.

**Comment by anxdpanic** [comment]
I can't see how youtube-dl or Context add-on could cause this other than an active background download(which I'm sure is already ruled out). 

Only other things I could think of for you; 

Are you using a proxy/vpn?
Check you haven't accidentally limited bandwidth in Kodi. Settings - System Settings - Internet Access - Internet connection bandwidth limitation

**Comment by dreamcat4** [comment]
Hi there thanks for these tips. Well those settings remain untouched and no, we are not using a proxy or any vpn like that. And no non-default artificial bandwidth limit setting within kodi.

But we do have a QoS bandwidth priority on our router. Which classes all downloads to this PC as bulk (lowest pri when other traffic). So I shall remove that and re-test.

The other thing is that our current internet is only 3.2 mbit which (although the issue being not fully utilized the whole link speed). Yet there are a few occasional reports from other people on forums who have exact same issue. Then they upgraded their internet speed to be faster, and its helped them to get enough for 720p quality on youtube.

It is not that there was enough already for 720p (the already is). But because there is some artificually limited cap (or maybe a fixed fraction % of the total available link speed), then having a higher maximum link speed might potentially also increase the capped youtube limit aswell. Or else otherwise get around a buggy speed negotiation protocol between client <--> server, that its incorrectly detects the real link speed (or makes too gross the intervals / increments).

Will find out next week when our broadband provider / ISP changes to the new one (its going to be 20x faster).

**Comment by dreamcat4** [comment]
OK have found the culprit. Apologise, sorry for incorrect bug report.

Last night setup new vpn feature in router settings, but it didnt work either. So to get this other feature working in the router we had to upgrade our router's firmware from Advanced Tomato v132 -->138. 

And this revealed a bug which exists in shibby's tomato versions less than v134. And is now fixed in v134.

From the changelog here:

http://tomato.groov.pl/?page_id=78

where it says:

```
[RELEASE] 134
ARM:
– Fix MARK module (BW Limiter, QoS and OVPN Routing policy should now works correct)
```

I'm pretty sure this was the issue ^^. Because indeed we had this feature (the QoS Bandwidth Limiter) enabled with a rule for that PC's local ip address, (to make it a lower 'bulk' priority). While we had the old router firmware, we saw this bug with an artificially limited youtube speed.

Since upgrading past shibby's tomato v134 yesterday (v138 now), then this youtube 720p buffering all works fine now and no problems. Yet did not change any of the router's settings in the QoS / BW Limiter. Its still the same user setting. And while we are still on the same existing 'slower' ISP at connection speed of 3.2 mbit/s.

Which all points to bug in router firmware that's all. Everything in the youtube all 720p videos playback very smoothly and responsively today. Almost never buffering or waiting, good like all the other devices in the house.

Many thanks for tips / help. Sorry for any inconvenience.
